Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
+
+					*/
+
+					throw oip::UnhandledException(surface);
+				}
+
+				/*! \brief  Converts \c IfcToroidalSurface to ...
+				\param	surface	\c IfcToroidalSurface entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcToroidalSurface is a subtype of \c IfcElementarySurface.
+				*/
+
+				void convertIfcToroidalSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcToroidalSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
+
+					/* TO DO: Finish implementation of convertIfcToroidalSurface.
+
+						// Get major and minor radius.
+						double majorRadius = surface->MajorRadius;
+						double minorRadius = surface->MinorRadius;
+
+					*/
+
+					throw oip::UnhandledException(surface);
+				}
+
+				/*! \brief  Converts \c IfcSurfaceOfLinearExtrusion to ...
+				\param	surface	\c IfcSurfaceOfLinearExtrusion entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcSurfaceOfLinearExtrusion is a subtype of \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurfaceOfLinearExtrusion(const EXPRESSReference<typename IfcEntityTypesT::IfcSurfaceOfLinearExtrusion>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
+
+					/* TO DO: Finish implementation of convertIfcSurfaceOfLinearExtrusion.
+
+					// Get swept curve and position.
+					EXPRESSReference<typename IfcEntityTypesT::IfcProfileDef> sweptSurfaceProfile = surface->SweptCurve;
+					EXPRESSReference<typename IfcEntityTypesT::IfcAxis2Placement3D> sweptSurfacePlacement = nullptr;
+
+					if (surface->Position) {
+						EXPRESSReference<typename IfcEntityTypesT::IfcAxis2Placement3D> ref = surface->Position;
+						sweptSurfacePlacement;
+					}
+					else {
+						BLUE_LOG(warning) << "#" << swept_surface->getId() << " IfcSweptSurface without placement found.";
+					}
+
+					carve::math::Matrix surfaceMatrix(pos);
+					if (sweptSurfacePlacement) {
+						surfaceMatrix = pos * placementConverter->convertIfcAxis2Placement3D(sweptSurfacePlacement);
+					}
+
+
+					// Get extrude direction and depth.
+					EXPRESSReference<typename IfcEntityTypesT::IfcDirection>& linearExtrusionDirection = surface->ExtrudedDirection;
+					double linearExtrusionDepth = surface->Depth;
+
+					*/
+
+					throw oip::UnhandledException(surface);
+
+				}
+
+				/*! \brief  Converts \c IfcSurfaceOfRevolution to ...
+				\param	surface	\c IfcSurfaceOfRevolution entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcSurfaceOfRevolution is a subtype of \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurfaceOfRevolution(const EXPRESSReference<typename IfcEntityTypesT::IfcSurfaceOfRevolution>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {

```

Comment:
Please convert `pos` and `polylineData` to return types and return them from the function.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1. On request by @pjanck only polylineData as return parameter.